src/backend/distributed/commands/foreign_constraint.c

```diff
@@ -561,106 +516,72 @@ get_relation_constraint_oid_compat(HeapTuple heapTuple)
+	foreach_oid(foreignKeyOid, foreignKeyOids)
 	{
-		Form_pg_constraint constraintForm = (Form_pg_constraint) GETSTRUCT(heapTuple);
+		HeapTuple heapTuple = SearchSysCache1(CONSTROID, ObjectIdGetDatum(
+												  foreignKeyOid));
 
-		if (constraintForm->contype != CONSTRAINT_FOREIGN)
-		{
-			heapTuple = systable_getnext(scanDescriptor);
-			continue;
-		}
+		Assert(HeapTupleIsValid(heapTuple));
 
-		Oid referencedTableId = constraintForm->confrelid;
+		Form_pg_constraint constraintForm = (Form_pg_constraint) GETSTRUCT(heapTuple);
+
+		Oid referencedTableOid = constraintForm->confrelid;
 
-		if (!IsCitusTable(referencedTableId))
+		if (!IsReferenceTable(referencedTableOid))
 		{
-			heapTuple = systable_getnext(scanDescriptor);
+			ReleaseSysCache(heapTuple);
 			continue;
 		}
 
-		if (PartitionMethod(referencedTableId) == DISTRIBUTE_BY_NONE)
-		{
-			hasForeignKeyToReferenceTable = true;
-			break;
-		}
+		fkeyOidsToReferenceTables = lappend_oid(fkeyOidsToReferenceTables, foreignKeyOid);
 
-		heapTuple = systable_getnext(scanDescriptor);
+		ReleaseSysCache(heapTuple);
 	}
 
-	/* clean up scan and close system catalog */
-	systable_endscan(scanDescriptor);
-	heap_close(pgConstraint, NoLock);
-	return hasForeignKeyToReferenceTable;
+	return fkeyOidsToReferenceTables;
 }
 
 
 /*
  * TableReferenced function checks whether given table is referenced by another table
- * via foreign constraints. If it is referenced, this function returns true. To check
- * that, this function searches for the given relation in the pg_constraint system
- * catalog table. However since there are no indexes for the column we search for,
- * this function performs sequential search. So call this function with caution.
+ * via foreign constraints. If it is referenced, this function returns true.
  */
 bool
-TableReferenced(Oid relationId)
+TableReferenced(Oid relationOid)
 {
-	ScanKeyData scanKey[1];
-	int scanKeyCount = 1;
-	Oid scanIndexId = InvalidOid;
-	bool useIndex = false;
-
-	Relation pgConstraint = heap_open(ConstraintRelationId, AccessShareLock);
-
-	ScanKeyInit(&scanKey[0], Anum_pg_constraint_confrelid, BTEqualStrategyNumber, F_OIDEQ,
-				relationId);
-	SysScanDesc scanDescriptor = systable_beginscan(pgConstraint, scanIndexId, useIndex,
-													NULL,
-													scanKeyCount, scanKey);
-
-	HeapTuple heapTuple = systable_getnext(scanDescriptor);
-
-	while (HeapTupleIsValid(heapTuple))
-	{
-		Form_pg_constraint constraintForm = (Form_pg_constraint) GETSTRUCT(heapTuple);
-
-		if (constraintForm->contype == CONSTRAINT_FOREIGN)
-		{
-			systable_endscan(scanDescriptor);
-			heap_close(pgConstraint, NoLock);
-
-			return true;
-		}
-
-		heapTuple = systable_getnext(scanDescriptor);
-	}
-
-	/* clean up scan and close system catalog */
-
-	systable_endscan(scanDescriptor);
-	heap_close(pgConstraint, NoLock);
+	int flags = EXTRACT_REFERENCED_CONSTRAINTS;
+	List *tableReferencedForeignConstraintOids = GetTableForeignConstraintOidsInternal(
```

Comment:
minor: Can you go over the  PR  and improve the indentations a bit, such as

```C
	List *tableReferencedForeignConstraintOids = GetTableForeignConstraintOidsInternal(
		relationOid, flags);
```
to
```C
	List *tableReferencedForeignConstraintOids = 	
		GetTableForeignConstraintOidsInternal(relationOid, flags);